mod/groups/classes/Elgg/Groups/Routes/Index.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+namespace Elgg\Groups\Routes;
+
+class Index implements \Elgg\Route {
+
+	public function __construct($identifier, $segments) {
+		$this->identifier = $identifier;
+		$this->segments = $segments;
+	}
+
+	public function validate() {
+		
+	}
+	
+	public function getBreadcrumbs() {
+		return [
+			[
+				'title' => elgg_echo('groups'),
+				'href' => 'groups/all',
+			],
+		];
+	}
+
+	public function getContent() {
+		if (elgg_get_plugin_setting('limited_groups', 'groups') != 'yes' || elgg_is_admin_logged_in()) {
+			elgg_register_title_button('groups', 'add', 'group');
+		}
+
+		$selected_tab = get_input('filter', 'newest');
+		if (!elgg_view_exists("groups/listing/$selected_tab")) {
+			$selected_tab = 'newest';
+		}
+		return elgg_view("groups/listing/$selected_tab");
```

Comment:
Ideally, we would do something like:

```php
return new \ElggList::factory($list_options);
```

Default viewtype will then resolve it to `elgg_list_entities()`, JSON will serialize the list, RSS will use whatever logic it has going.